### DISCUSSION TITLE
Allow swizzle field names in array initialization

### DISCUSSION THREAD

**Discussion by Porem5002** [odin-lang/Odin#4289] [Ideas/Requests]
Issue: #4288

This feature comes from something that was previously possible using the raylib bindings.

Until around [dev-2023-11](https://github.com/odin-lang/Odin/releases/tag/dev-2023-11), the Color type from raylib was defined as a struct so we could write code like this and it was pretty obvious what it would do:
```odin
x: raylib.Color = { r = 127, g = 0, b = 127, a = 255  }
```
But since then (after [this](https://github.com/odin-lang/Odin/pull/2927) more precisely) this type is defined as a fixed array, which allows for more flexibility but also means that the previous code no longer compiles since those fields don't actually exist in an array.

My request is to allow for this kind of syntax for initializing arrays, making the language more consistent.

**Comment by flysand7** [comment]
I'm pretty sure we have an array initializer that allows specific indices for values:

```odin
a := [?]string {
  0 = "pawn",
  3 = "knight",
  10 = "rook",
}
```

In which case it would make sense if this syntax was allowed for vector initialization.

However one thing that I'm curious is whether that'll conflict with enum-array initialization, in particular where:

```odin
E :: enum { x, y, z, w }

a := [E]int {
  x = 1,
  y = 2,
  z = 3,
  w = 4,
}
```

Not making any point, just bringing up ideas that may hinder the viability of implementation.
